apps/scene2pset/scene2pset.cc

```diff
@@ -43,8 +44,76 @@ struct AppSettings
     float min_valid_fraction = 0.0f;
     float scale_factor = 2.5f; /* "Radius" of MVS patch (usually 5x5). */
     std::vector<int> ids;
+    bool output_correspondence = false;
 };
 
+struct CorrespondenceViewMetadata
+{
+    unsigned int view_id; 
+    unsigned int width;  // After downsampling. 
+    unsigned int height; // After downsampling.
+    unsigned long first_idx;
+
+    CorrespondenceViewMetadata (unsigned int view_id, unsigned int width, unsigned int height, unsigned long first_idx):
+        view_id(view_id), width(width), height(height), first_idx(first_idx) {}
+
+};
+
+struct CorrespondenceData
+{
+    std::vector<math::Vec2ui> data;
+    std::vector<CorrespondenceViewMetadata> metadata;
+};
+
+void
+append_correspondence_data_from_view(CorrespondenceData& corr_data, 
```

Comment:
Don't pass non-const references. If you want to modify a variable, pass it as last argument as pointer. MVE may not do this consistently, but it's good practice. This way it's explicit at the caller site that a variable gets modified.